### ISSUE TITLE
Ambiguous (N) basepairs in contigs.fasta output

### ISSUE THREAD

**Issue by chrisgulvik** [ablab/spades#273]
## Question
When I updated from v3.9.0, I noticed quite often the __contigs.fasta__ output file now includes 100 bp segments of ambiguous (N) nucleotides. In all cases with my 2x250 bp input, the N stretches only show up in the final K127 subdir, not smaller k-mer sizes. The manual online (http://cab.spbu.ru/files/release3.12.0/manual.html) indicates _"if two contigs are separated by a complex tandem repeat, that cannot be resolved exactly, contigs are joined into scaffold with a fixed gap size of 100 bp. Contigs produced by SPAdes do not contain N symbols."_ and far as I can remember this has always been the case, but is this a recent (permanent) change you guys made?

- v3.9.0 : no 'NN' within contigs.fasta
- v3.12.0 : multiple 100 bp 'N' stretches in contigs.fasta

This occurs with or without read cleaning, so mine as well skip it to simplify here. The commands below should reproduce the Ns within the contigs.fasta output file.

## Example
```
wget ftp://ftp-trace.ncbi.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR777/SRR7774417/SRR7774417.sra
fastq-dump -O $PWD --dumpbase --split-files --readids -Q 33 \
  -defline-qual '+' --defline-seq '@$ac_$sn[_$rn]/$ri' \
  SRR7774417.sra
	#	Read 776523 spots for SRR7774417.sra
	#	Written 776523 spots for SRR7774417.sra
du -sh SRR7774417_[12].fastq
	#	391M	SRR7774417_1.fastq
	#	391M	SRR7774417_2.fastq

spades.py -1 SRR7774417_1.fastq -2 SRR7774417_2.fastq -o v3.12.0 --only-assembler
grep -n 'NN' v3.12.0/contigs.fasta
	#	contigs.fasta:22320:TTCGGCGTTCGGCGTTCGGCGTTCGGCGTTCGGCGNNNNNNNNNNNNNNNNNNNNNNNNN
	#	contigs.fasta:22321:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN
	#	contigs.fasta:22322:NNNNNNNNNNNNNNNTTCAGCGTTCAGCGTTCAGCGTTCAGCGTTCAGCGTTCAGCGTTC
	#	contigs.fasta:27706:GCCGAGCGCCGAGCGCCGAGCGCNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN
	#	contigs.fasta:27707:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN
	#	contigs.fasta:27708:NNNCAAGCGCCAAGCGCCAAGCGCCAAGCGCCAAGCGCCAAGCGCCAAGCGCCAAGCGCC
	#	contigs.fasta:31992:GGTCGGNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN
	#	contigs.fasta:31993:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNCAAGCATGAACTTC
	#	contigs.fasta:87047:GAGCTTCGAGCTTCGAGCTTCGAGCTTCGAGCTTCGAGCTTCGANNNNNNNNNNNNNNNN
	#	contigs.fasta:87048:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN
	#	contigs.fasta:87049:NNNNNNNNNNNNNNNNNNNNNNNNGCTCGATTCGACCACTCCGCCGAATCAATGCCCGAA
	#	contigs.fasta:107434:GTCTCCGTCTCCGTCTCCGTCTCCGTCTCCGTCTCCGTCTCCGTCTNNNNNNNNNNNNNN
	#	contigs.fasta:107435:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN
	#	contigs.fasta:107436:NNNNNNNNNNNNNNNNNNNNNNNNNNCCATCAGCCATCAGCCATCAGCCATCAGCCGCCT
	#	contigs.fasta:112790:CTTGCTGCTTGCTGNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN
	#	contigs.fasta:112791:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNCTTGCC

grep -l 'NN' v3.12.0/K*/*.fasta
	#	v3.12.0/K127/final_contigs.fasta
	#	v3.12.0/K127/scaffolds.fasta
```

**Comment by asl** [comment]
Hello

I believe we fixed this issue in SPAdes 3.13.0. Will you please give it a try and verify if this issue is fixed there?

**Comment by chrisgulvik** [comment]
v3.12.0 is the newest we have on our cluster, but I'll request the update and check back. Thanks for the speedy response!

**Comment by asl** [comment]
Actually, fixed in 3.12.1 (https://github.com/ablab/algorithmic-biology/commit/72f453a9c89a2d71c7a5570c37e54e10b5794a7d)
